docs/csharp/tutorials/mixins-with-default-interface-methods.md

```diff
@@ -0,0 +1,122 @@
+
+In this tutorial, you'll learn how to:
+
+> [!div class="checklist"]
+>
+> * Create interfaces with implementations that describe discrete features.
+> * Create classes that use the default implementations.
+> * Create classes that override some or all of the default implementations.
+> * Expose default interface methods to classes using the `virtual` keyword.
+
+## Prerequisites
+
+You’ll need to set up your machine to run .NET Core, including the C# 8.0 preview compiler. The C# 8.0 preview compiler is available starting with [Visual Studio 2019](https://visualstudio.microsoft.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2019), or the latest [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.0). Default interface members are available beginning with .NET Core 3.0 preview 4.
+
+## Limitations of extension methods
+
+One way you can implement behavior that appears as part of an interface is to define [extension methods](../programming-guide/classes-and-structs/extension-methods.md) that provide the default behavior. Interfaces declare a minimum set of members while providing a greater surface area for any class that implements that interface. For example, the extension methods in <xref:System.Linq.Enumerable> provide the implementation for any sequence to be the source of a LINQ query.
+
+Extension methods are resolved at compile time, using the declared type of the variable. Classes that implement the interface can provide a better implementation for any extension method. Variable declarations must match the implementing type to enable the compiler to choose that implementation. When the compile-time type matches the interface, method calls resolve to the extension method. Another concern with extension methods is that those methods are accessible wherever the class containing the extension methods is accessible. Classes cannot declare if they should or should not provide features declared in extension methods.
+
+Starting with C# 8.0, you can declare the default implementations as interface methods. Then, every class automatically uses the default implementation. Any class that can provide a better implementation can override the interface method definition with a better algorithm. In one sense, this technique sounds similar to how you could use [extension methods](../programming-guide/classes-and-structs/extension-methods.md). Default interface methods can be declared with the `virtual` keyword, which means those methods become part of the interface of implementing classes. WIthout the `virtual` keyword on the interface method, default interface methods are accessible only with a reference to the interface type.
```

Comment:
Nit: `WIthout`.

Super sorry for being persistent, but maybe you can help me clear up possible misunderstandings of the terminology on my side.

> WIthout the `virtual` keyword on the interface method, default interface methods are accessible only with a reference to the interface type.

From my understanding of reading this it would suggest that by adding the `virtual` keyword should make the commented out line in the following code compile (by interpreting the inverse):
```csharp
public interface ICanReturnMyName
{
    public virtual string MyName() => nameof(ICanReturnMyName);
}
public class NameReturner : ICanReturnMyName {}
class Program
{
    static void Main(string[] args)
    {
        NameReturner nr = new NameReturner();
        ICanReturnMyName iNr = nr;
        // next line doesn't compile
        //Console.WriteLine($"NameReturner:         {nr.MyName()}");
        Console.WriteLine($"ICanReturnMyName iNr: {iNr.MyName()}");
    }
}
```

Reply:
@dasMulli You are correct. Thanks for catching this. I honestly don't know what I was looking at when I tested this.  Fixing now.